spec/system/admin/orders_spec.rb

```diff
@@ -621,10 +624,134 @@ def extract_pdf_content
             end
           end
 
-          it_behaves_like "can bulk print invoices from 2 orders"
+          shared_examples "should ignore the non invoiceable order" do
+            it "bulk prints invoices in pdf format" do
+              page.find(order4_selector).click
+              page.find(order5_selector).click
+
+              page.find("span.icon-reorder", text: "ACTIONS").click
+              within ".ofn-drop-down .menu" do
+                expect {
+                  page.find("span", text: "Print Invoices").click # Prints invoices in bulk
+                }.to enqueue_job(BulkInvoiceJob).exactly(:once)
+              end
+
+              expect(page).to have_content "Compiling Invoices"
+              expect(page).to have_content "Please wait until the PDF is ready " \
+                                           "before closing this modal."
+
+              perform_enqueued_jobs(only: BulkInvoiceJob)
+
+              expect(page).to have_content "Bulk Invoice created"
 
-          context "with legal invoices feature", feature: :invoices do
+              within ".modal-content" do
+                expect(page).to have_link(class: "button", text: "VIEW FILE",
+                                          href: /invoices/)
+
+                invoice_content = extract_pdf_content
+
+                expect(invoice_content).to have_content("TAX INVOICE", count: 1)
+                expect(invoice_content).to_not have_content(order4.number.to_s)
+                expect(invoice_content).to have_content(order5.number.to_s)
+                expect(invoice_content).to_not have_content(distributor4.name.to_s)
+                expect(invoice_content).to have_content(distributor5.name.to_s)
+                expect(invoice_content).to_not have_content(order_cycle4.name.to_s)
+                expect(invoice_content).to have_content(order_cycle5.name.to_s)
+              end
+            end
+          end
+
+          context "ABN is not required" do
+            before do
+              allow(Spree::Config).to receive(:enterprise_number_required_on_invoices?)
+                .and_return false
+            end
             it_behaves_like "can bulk print invoices from 2 orders"
+
+            context "with legal invoices feature", feature: :invoices do
+              it_behaves_like "can bulk print invoices from 2 orders"
+            end
+
+            context "one of the two orders is not invoiceable" do
+              before do
+                order4.cancel!
+                assert(!order4.invoiceable?)
+                assert(order5.invoiceable?)
```

Comment:
Having assertion in the `before` block somehow doesn't sit well with me, and it's weird to see `minitest` syntax in a spec file.
I would move this to it's own `it` block :
```suggestion
it "checks orders are properly setup" do
  expect(order4.invoiceable?).to be(false)
  expect(order5.invoiceable?).to be(true)
end
```

But again you are just checking the data is setup properly, which I think isn't necessary you should know in what state the data is before the test runs, so I would just remove the assert here.